### PR TITLE
Adding analyzer for ``need for speed`` concept exercise

### PR DESCRIPTION
**Pull request by manumafe98**
closes [#109](https://github.com/exercism/java-analyzer/issues/109)

To do:

- [x] Merge https://github.com/exercism/website-copy/pull/2325

### CODE REVIEW

**Comment by manumafe98** [comment]
I have a couple of doubts of this particular case, cause this exercise has two classes, so I need to create an analyzer for ``RaceTrack `` ? I would like your tips @sanderploegsma to know if I'm in the correct path

**Comment by sanderploegsma** [comment]
Overall this looks good!

> I have a couple of doubts of this particular case, cause this exercise has two classes, so I need to create an analyzer for `RaceTrack ` ?

No there is no need to create separate analyzers for each class. The analyzer is invoked with a list of `CompilationUnit`s, which contain all the classes in the solution. Each `CompilationUnit` corresponds to a Java file, FYI.

If your analyzer needs to have some logic specific to a single class of a multi-class solution, there are multiple ways of achieving this. Here's one from the top of my head:

```java
public class NeedForSpeedAnalyzer implements Analyzer {
    @Override
    public void analyze(Solution solution, OutputCollector output) {
        var needForSpeedClassAnalyzer = new NeedForSpeedClassAnalyzer();
        var raceTrackClassAnalyzer = new RaceTrackClassAnalyzer();

        for (var compilationUnit : solution.getCompilationUnits()) {
            compilationUnit.getClassByName("NeedForSpeed").ifPresent(c -> c.accept(needForSpeedClassAnalyzer, output));
            compilationUnit.getClassByName("RaceTrack").ifPresent(c -> c.accept(raceTrackClassAnalyzer, output));
        }
    }
}

class NeedForSpeedClassAnalyzer extends VoidVisitorAdapter<OutputCollector> {
    @Override
    public void visit(MethodDeclaration node, OutputCollector output) {
        // Only called for methods in the NeedForSpeed class
    }
}

class RaceTrackClassAnalyzer extends VoidVisitorAdapter<OutputCollector> {
    @Override
    public void visit(MethodDeclaration node, OutputCollector output) {
        // Only called for methods in the RaceTrack class
    }
}
```

Hope that helps!

File: src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package analyzer.exercises.needforspeed;
+
+import analyzer.Comment;
+
+public class AvoidConditionalLogic extends Comment {
+    @Override
+    public String getKey() {
+        return "java.needforspeed.avoid_conditional_logic";
```
Comment (sanderploegsma) [inline src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java:8]: The key defines the path in the website-copy repository where the template is located, in this case that would be `java/needforspeed/avoid_conditional_logic.md`. 

I would prefer if we store the templates in directories named after the exercise slug, so `java/need-for-speed/avoid_conditional_logic.md`, which means they key here should be `java.need-for-speed.avoid_conditional_logic`.

The same goes for the other comments as well.

File: src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package analyzer.exercises.needforspeed;
+
+import analyzer.Comment;
+
+public class AvoidConditionalLogic extends Comment {
```
Comment (sanderploegsma) [inline src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java:5]: Can you make the comments here package-private? That makes it a little harder for other exercise analyzers to accidentally reuse them.

Comment (sanderploegsma) [inline src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java:5]: Please follow the convention used in the other exercises, where a bit of Javadoc is used to link to the corresponding Markdown template in the website-copy repository.

File: src/test/java/analyzer/AnalyzerIntegrationTest.java [hunk 3]
```diff
@@ -97,4 +97,20 @@ public void twofer(String scenario) throws IOException {
...
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "ExemplarSolution",
+            "UsingForLoop",
+            "UsingIfStatement",
+            "UsingTernary",
+            "UsingWhileLoop",
+    })
+    void needforspeed(String scenario) throws IOException {
+        var path = Path.of("needforspeed", scenario + ".java");
```
Comment (sanderploegsma) [inline src/test/java/analyzer/AnalyzerIntegrationTest.java:110]: Can you change the path of the scenarios to use the slug instead (so `need-for-speed`)?

File: src/main/java/analyzer/AnalyzerRoot.java [hunk 4]
```diff
@@ -49,6 +50,7 @@ private static List<Analyzer> createAnalyzers(String slug) {
             case "lasagna" -> analyzers.add(new LasagnaAnalyzer());
             case "leap" -> analyzers.add(new LeapAnalyzer());
             case "two-fer" -> analyzers.add(new TwoferAnalyzer());
+            case "need-for-speed" -> analyzers.add(new NeedForSpeedAnalyzer());
```
Comment (sanderploegsma) [inline src/main/java/analyzer/AnalyzerRoot.java:53]: Can you make sure the switch cases are ordered alphabetically?

File: src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java (hunk 2 above)
Comment (manumafe98) [reply src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java]: Done!

Comment (manumafe98) [reply src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java]: Done!

File: src/main/java/analyzer/AnalyzerRoot.java (hunk 4 above)
Comment (manumafe98) [reply src/main/java/analyzer/AnalyzerRoot.java:52]: Done!

File: src/test/java/analyzer/AnalyzerIntegrationTest.java (hunk 3 above)
Comment (manumafe98) [reply src/test/java/analyzer/AnalyzerIntegrationTest.java]: Done!

File: src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java (hunk 1 above)
Comment (manumafe98) [reply src/main/java/analyzer/exercises/needforspeed/AvoidConditionalLogic.java]: Done!

File: src/main/java/analyzer/exercises/needforspeed/NeedForSpeedAnalyzer.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import analyzer.comments.ExemplarSolution;
+
+/**
+ * The {@link NeedForSpeedAnalyzer} is the analyzer implementation for the {@code need-for-speed} practice exercise.
+ * It has two subclasses NeedForSpeedClassAnalyzer and RaceTrackClassAnalyzer that extend the {@link VoidVisitorAdapter} and use the visitor pattern to traverse each compilation unit.
+ * 
+ * @see <a href="https://github.com/exercism/java/tree/main/exercises/concept/need-for-speed">The need-for-speed exercise on the Java track</a>
+ */
+public class NeedForSpeedAnalyzer implements Analyzer {
+    private static final String EXERCISE_NAME = "NeedForSpeed";
```
Comment (sanderploegsma) [inline src/main/java/analyzer/exercises/needforspeed/NeedForSpeedAnalyzer.java:21]: This string will be rendered into an analyzer comment, which means it's part of the text shown to students. It's supposed to contain the name of the exercise as it's shown on the website, so change it to:

```suggestion
    private static final String EXERCISE_NAME = "Need for Speed";
```

**Review by sanderploegsma — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]

**Comment by SleeplessByte** [comment]
Analyzer comments have been merged into `website-copy`

**Pull request merged**